waterspout_api/serializers.py

```diff
@@ -78,6 +82,82 @@ def update(self, instance, validated_data):
 		return instance
 
 
+class EmailSerializer(serializers.Serializer):
+	email = serializers.EmailField()
+
+	class Meta:
+		fields = ("email",)
+
+
+class ResetPasswordSerializer(serializers.Serializer):
+	password = serializers.CharField(
+		write_only=True
+	)
+
+	class Meta:
+		fields = ("password",)
+
+	def validate(self, data):
+
+		# user's new password
+		password = data.get("password")
+		kwargs = self.context.get("kwargs", {})
+		token = kwargs.get("token")
+		encoded_pk = kwargs.get("encoded_pk")
+
+		if token is None:
+			raise serializers.ValidationError("Missing token")
+		elif encoded_pk is None:
+			raise serializers.ValidationError("Missing encoded_pk")
+
+		pk = urlsafe_base64_decode(encoded_pk).decode()
+
+		user = get_user_model().objects.get(pk=pk)
+		if not user:
+			raise serializers.ValidationError("User PK not found")
+
+		if PasswordResetTokenGenerator().check_token(user, token):
+			user.set_password(password)
+			user.save()
+			return data
+		else:
+			raise serializers.ValidationError("Token is not valid")
+
+
+class ChangePasswordSerializer(serializers.Serializer):
+	password = serializers.CharField(
+		write_only=True
+	)
+
+	class Meta:
+		model = get_user_model()
+		fields = ("password", "token", "old_password")
+
+	def validate(self, data):
+		kwargs = self.context.get("kwargs", {})
+		token = kwargs.get("token")
+		old_password = kwargs.get("old_password")
+
+		if token is None:
+			raise serializers.ValidationError("Missing token")
+		elif old_password is None:
+			raise serializers.ValidationError("Missing encoded_pk")
+
+		if self.Meta.model.objects.filter(auth_token=token) != "":  # Check if user is signed in
+			user = self.Meta.model.objects.get(auth_token=token)  # Get user using token
```

Comment:
Same comment as above - we may want to move the actual DB-modifying actions to the view.